src/map/mercenary.cpp

```diff
@@ -474,137 +449,447 @@ void mercenary_kills(struct mercenary_data *md){
+	if (!md || !md->db || md->db->skill.count(skill_id) == 0)
 		return 0;
-	return md->db->skill[idx].lv;
+	return md->db->skill[skill_id];
 }
 
-/**
-* Read each line of Mercenary's database
-**/
-static bool mercenary_readdb_sub(char* str[], int columns, int current)
-{
-	int ele;
-	uint16 class_ = atoi(str[0]);
-	struct s_mercenary_db *db;
-	struct status_data *status;
-
-	db = &mercenary_db_data[class_];
-
-	db->class_ = class_;
-	safestrncpy(db->sprite, str[1], NAME_LENGTH);
-	safestrncpy(db->name, str[2], NAME_LENGTH);
-	db->lv = atoi(str[3]);
-
-	status = &db->status;
-	db->vd.class_ = db->class_;
-
-	status->max_hp = atoi(str[4]);
-	status->max_sp = atoi(str[5]);
-	status->rhw.range = atoi(str[6]);
-	status->rhw.atk = atoi(str[7]);
-	status->rhw.atk2 = status->rhw.atk + atoi(str[8]);
-	status->def = atoi(str[9]);
-	status->mdef = atoi(str[10]);
-	status->str = atoi(str[11]);
-	status->agi = atoi(str[12]);
-	status->vit = atoi(str[13]);
-	status->int_ = atoi(str[14]);
-	status->dex = atoi(str[15]);
-	status->luk = atoi(str[16]);
-	db->range2 = atoi(str[17]);
-	db->range3 = atoi(str[18]);
-	status->size = atoi(str[19]);
-	status->race = atoi(str[20]);
-
-	ele = atoi(str[21]);
-	status->def_ele = ele%20;
-	status->ele_lv = (unsigned char)floor(ele/20.);
-	if( !CHK_ELEMENT(status->def_ele) )
-	{
-		ShowWarning("Mercenary %d has invalid element type %d (max element is %d)\n", db->class_, status->def_ele, ELE_ALL - 1);
-		status->def_ele = ELE_NEUTRAL;
+const std::string MercenaryDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/mercenary_db.yml";
+}
+
+/*
+ * Reads and parses an entry from the mercenary_db.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
+ */
+uint64 MercenaryDatabase::parseBodyNode(const YAML::Node &node) {
+	uint32 id;
+
+	if (!this->asUInt32(node, "Id", id))
+		return 0;
+
+	std::shared_ptr<s_mercenary_db> mercenary = this->find(id);
+	bool exists = mercenary != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "AegisName", "Name" }))
+			return 0;
+
+		mercenary = std::make_shared<s_mercenary_db>();
+		mercenary->class_ = id;
 	}
-	if( !CHK_ELEMENT_LEVEL(status->ele_lv) )
-	{
-		ShowWarning("Mercenary %d has invalid element level %d (max is %d)\n", db->class_, status->ele_lv, MAX_ELE_LEVEL);
-		status->ele_lv = 1;
+
+	if (this->nodeExists(node, "AegisName")) {
+		std::string name;
+
+		if (!this->asString(node, "AegisName", name))
+			return 0;
+
+		if (name.size() > NAME_LENGTH) {
+			this->invalidWarning(node["AegisName"], "%s's size exceeds %d, skipping.\n", name.c_str(), NAME_LENGTH);
+			return 0;
+		}
+
+		safestrncpy(mercenary->sprite, name.c_str(), sizeof(mercenary->sprite));
+	}
+
+	if (this->nodeExists(node, "Name")) {
+		std::string name;
+
+		if (!this->asString(node, "Name", name))
+			return 0;
+
+		safestrncpy(mercenary->name, name.c_str(), sizeof(mercenary->name));
```

Comment:
```suggestion
		if (name.size() > NAME_LENGTH) {
			this->invalidWarning(node["Name"], "Name \"%s\" exceeds maximum of %d characters, capping...\n", name.c_str(), NAME_LENGTH - 1);
		}

		name.resize(NAME_LENGTH);
		safestrncpy(mercenary->name, name.c_str(), sizeof(mercenary->name));
```

Also, convert to std::string!